server/channels/app/oauth.go

```diff
@@ -822,16 +822,18 @@ func (a *App) AuthorizeOAuthUser(c request.CTX, w http.ResponseWriter, r *http.R
 	stateEmail := stateProps["email"]
 	stateAction := stateProps["action"]
 	if stateAction == model.OAuthActionEmailToSSO && stateEmail == "" {
+		c.Logger().Error("No email provided in state when trying to switch from email to SSO")
 		return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest)
```

Comment:
I wonder if it's even better to wrap a new error and return it:

```suggestion
		err := errors.New("No email provided in state when trying to switch from email to SSO")
		return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest).Wrap(err)
```

That way a single line the logs contains all the context on what went wrong.

Reply:
Yeah I like it, done